shared/sub_plugin.go

```diff
@@ -0,0 +1,61 @@
+package shared
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+)
+
+type AttributedObservation[ObservationType any] struct {
+	OracleID    commontypes.OracleID
+	Observation ObservationType
+}
+
+// SubPlugin This interface is to create multiple SubPlugins/Processors under OCR plugin.
+// The idea is to separate the logic of different types of observations and outcomes into separate processors.
+// This makes it easier to manage and test the logic of each type of observation/outcome without affecting each others
+// Some of them will implement state machines (e.g. merkleroot), others might implement simpler logic. (e.g. token)
+// The OCR plugin becomes a coordinator/collector of these SubPlugins.
+// Example Pseudo code:
+//
+//	OCRPlugin {
+//	  nodeID
+//	  merkleProcessor
```

Comment:
renames
`merkleRootsProcessor, tokenPricesProcessor, chainFeeProcessor`

Reply:
Also rename `OCRPLugin.Observer` to `OCRPlugin.Observation` and probably this example is enough, no need for validate and outcome